porcupine/plugins/directory_tree.py

```diff
@@ -381,16 +381,33 @@ def _populate_contextmenu(self) -> None:
         self.contextmenu.delete(0, "end")
         self.event_generate("<<PopulateContextMenu>>")
 
-    def _on_right_click(self, event: tkinter.Event[DirectoryTree]) -> str | None:
+    def _on_right_click(self, event: tkinter.Event[DirectoryTree], menu_key: bool) -> str | None:
         self.tk.call("focus", self)
 
-        item: str = self.identify_row(event.y)
-        self.set_the_selection_correctly(item)
+        # Behaviour for menu/application key
+        if menu_key:
+            # Get relative position of focus item
+            x, y, width, height = self.bbox(self.focus())
```

Comment:
Currently the mypy check is failing with this mypy error:

```
porcupine/plugins/directory_tree.py:390: error: "Literal['']" object is not iterable  [misc]
```

This points out a real problem: if you have so many items in the treeview that not all of them fit on the screen, you select an invisible item, and then you press the menu key, you get an error:

```
porcupine._state ERROR: Error in tkinter callback
Traceback (most recent call last):
  File "/usr/lib/python3.9/tkinter/__init__.py", line 1892, in __call__
    return self.func(*args)
  File "/home/akuli/porcu/porcupine/plugins/directory_tree.py", line 390, in _on_right_click
    x, y, width, height = self.bbox(self.focus())
ValueError: not enough values to unpack (expected 4, got 0)
```

In this case, `self.bbox()` returns an empty string, and so it's trying to do `x, y, width, height = ""`, and that fails. You need to add an if statement to check that.

Once you have fixed this, you need to run `black porcupine/plugins/directory_tree.py` and commit the result. Then this PR should be done :)

Reply:
Committed 4783f94d77caedfd01c711c5d9831de334cdf062.

BTW on a similar note, for me, right-clicking outside of a tree item raises errors such as:
```
  File "...\porcupine\porcupine\plugins\hide_project.py", line 13, in add_hide_option
    [project_id] = tree.selection()
ValueError: not enough values to unpack (expected 1, got 0)
```
```
  File ""...\porcupine\porcupine\plugins\python_venv.py", line 121, in _populate_menu
    [item] = tree.selection()
ValueError: not enough values to unpack (expected 1, got 0)
```
```
  File ""...\porcupine\porcupine\plugins\git_right_click.py", line 33, in populate_menu
    [item] = tree.selection()
ValueError: not enough values to unpack (expected 1, got 0)
```

These can be fixed with a try/except, as done in `porcupine\plugins\directory_tree.py` lines 318-322 and lines 358-361. 
Thoughts?

Reply:
Make a separate issue about your other errors :)